File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int pairSum(ListNode* head) {
        
        ListNode* tempNode = head;
        
        std::stack<int> stack;
        
        int n = 0;
        
        while(head != nullptr){
            stack.push(head->val);
            head = head->next;
            n++;
        }
        
        int maxSum = 0;
        
        for(int i = 0; i < n / 2; ++i){
            int twinValue = stack.top();
            stack.pop();

            // Calculate twin sum and update maxSum
            int twinSum = twinValue + tempNode->val;
            if (twinSum > maxSum) {
                maxSum = twinSum;
            }

            // Move to the next node
            tempNode = tempNode->next;
        }
        
        return maxSum;
        
    }
};

File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int pairSum(ListNode head) {
        
        int max = 0;
        
        List<Integer> list = new ArrayList<>();
        
        while(head != null){
            
            list.add(head.val);
            
            head = head.next;
        }
        
        for(int i = 0;i<list.size()/2;i++){
            
            max = Math.max(max,list.get(i) + list.get(list.size() - i - 1));
            
        }
        
        return max;
    }
}